src/NuGetForUnity/Editor/NugetWindow.cs

```diff
@@ -724,11 +746,24 @@ private void DrawUpdatesHeader()
                         UpdateUpdatePackages();
                     }
 
-                    if (GUILayout.Button("Install All Updates", GUILayout.Width(150)))
+                    if (updatePackages.Count > 0)
                     {
-                        NugetHelper.UpdateAll(updatePackages, NugetHelper.InstalledPackages);
-                        NugetHelper.UpdateInstalledPackages();
-                        UpdateUpdatePackages();
+                        if (GUILayout.Button("Install All Updates", GUILayout.Width(150)))
+                        {
+                            NugetHelper.UpdateAll(updatePackages, NugetHelper.InstalledPackages);
+                            NugetHelper.UpdateInstalledPackages();
+                            UpdateUpdatePackages();
+                        }
+                    }
+
+                    if (NugetHelper.SelectedPackagesCount > 0)
+                    {
+                        if (GUILayout.Button("Install All Selected Updates", GUILayout.Width(200)))
+                        {
+                            NugetHelper.UpdateAllSelected(updatePackages);
```

Comment:
It looks like you didn't have to add new UpdateAllSelected method. You could have just called this:
```c#
NugetHelper.UpdateAll(selectedInstalledPackages, NugetHelper.InstalledPackages);
NugetHelper.UnselectAll();
```